mono/mini/interp/interp.c

```diff
@@ -735,15 +734,29 @@ static InterpMethodArguments* build_args_from_sig (MonoMethodSignature *sig, Mon
 		case MONO_TYPE_CLASS:
```

Comment:
Instead of writing these trampolines, you might want to use gsharedvt-in wrappers to call icalls, they don't require any arch specific code.